File: search-insert-position/search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
 var searchInsert = function(nums, target) {
    
    let begin = 0;
    let end = nums.length - 1;
    let middle;
    
    while(begin <= end)
    {
        middle = Math.floor((begin+end)/2);
        if(nums[middle] < target)
        {
            begin = middle + 1;       
        }
        else if(nums[middle] > target)
        {
            end = middle - 1;            
        }
        else
        {
            return middle;
        }
    }
    return end+1;
};

File: search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        
        int first = 0, middle;
        int last = nums.length -1;
        
        while(first<=last)
        {
            middle = Math.round((first + last)/2);
            if(nums[middle] < target)
            {
                first = middle + 1; 
            }
            else if(nums[middle] > target)
            {
                last = middle - 1;
            }
            else 
            {
                return middle;
            }
        }
        return last + 1; 
    }
}
